File: app/src/main/java/com/cn29/aac/ui/feedentry/vm/FeedEntryListViewModelFactory.java
package com.cn29.aac.ui.feedentry.vm;

import com.cn29.aac.repo.feedentry.FeedEntryRepository;

import javax.inject.Inject;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;


/**
 * Created by Charles Ng on 20/9/2017.
 */

public class FeedEntryListViewModelFactory extends ViewModelProvider.NewInstanceFactory {
  private FeedEntryRepository feedEntryRepository;

  @Inject
  public FeedEntryListViewModelFactory( FeedEntryRepository feedEntryRepository) {
    this.feedEntryRepository = feedEntryRepository;
  }

  @Override
  public <T extends ViewModel> T create(Class<T> modelClass) {
    return (T) new FeedEntryListViewModel(feedEntryRepository);
  }
}

File: app/src/main/java/com/cn29/aac/ui/feedentry/vm/FeedEntryListViewModelFactory.kt
package com.cn29.aac.ui.feedentry.vm

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import com.cn29.aac.repo.feedentry.FeedEntryRepository
import javax.inject.Inject

/**
 * Created by Charles Ng on 20/9/2017.
 */
class FeedEntryListViewModelFactory @Inject constructor(private val feedEntryRepository: FeedEntryRepository) :
        NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return FeedEntryListViewModel(feedEntryRepository) as T
    }

}